modules/_labs/select/react/lib/SelectMenu.tsx

```diff
@@ -133,21 +188,152 @@ const MenuList = styled('ul')<Pick<SelectProps, 'error' | 'theme'>>(
   })
 );
 
+const generatePopperOptions = (
+  props: Pick<SelectMenuProps, 'isFlipped' | 'menuRef' | 'shouldAutoFlip' | 'shouldAutoFocus'>
+) => {
+  const {isFlipped, menuRef, shouldAutoFlip, shouldAutoFocus} = props;
+
+  let fallbackPlacements: Placement[] = [];
+  if (shouldAutoFlip) {
+    fallbackPlacements = isFlipped ? ['bottom'] : ['top'];
+  }
+
+  const modifiers = [
+    {
+      name: 'flip',
+      options: {
+        fallbackPlacements,
+      },
+    },
+    {
+      name: 'offset',
+      options: {
+        offset: () => {
+          const skidding = 0;
+
+          // Displace menu towards the button to obscure the bottom
+          // edge of the button and to create a smooth visual
+          // connection between the button and the menu
+          const distance = -parseInt(borderRadius.m, 10);
+
+          return [skidding, distance];
+        },
+      },
+    },
+    {
+      name: 'preventOverflow',
+      options: {
+        // Ensure the menu stays aligned with its reference (button),
+        // even if that means the menu is pushed out of view
+        mainAxis: false,
+      },
+    },
+  ];
+
+  return {
+    modifiers,
+    onFirstUpdate: () => {
+      if (shouldAutoFocus && menuRef && menuRef.current) {
+        menuRef.current.focus();
+      }
+    },
+  };
+};
+
 const SelectMenu = (props: SelectMenuProps) => {
-  const {children, error, isAnimated, isHiding, menuRef, ...elemProps} = props;
-
-  return (
-    <Menu error={error} isAnimated={isAnimated} isHiding={isHiding}>
-      <MenuList error={error} ref={menuRef} role="listbox" tabIndex={-1} {...elemProps}>
-        {children}
-      </MenuList>
-    </Menu>
-  );
+  const {
+    buttonRef,
+    children,
+    error,
+    isFlipped,
+    isHidden,
+    isHiding,
+    menuRef,
+    onCloseOnEscape,
+    shouldAnimate,
+    shouldAutoFlip,
+    shouldAutoFocus,
+    ...elemProps
+  } = props;
+
+  const popupRef = React.useRef<HTMLDivElement>(null);
+
+  const [width, setWidth] = useState(0);
+
+  const handleWidthChange = () => {
+    if (buttonRef.current) {
+      const newMenuWidth = buttonRef.current.clientWidth + 2 * buttonBorderWidth;
+      setWidth(newMenuWidth);
+    }
+  };
+
+  useLayoutEffect(() => {
+    handleWidthChange();
+  }, [buttonRef]);
```

Comment:
An optimization here would be to only measure the width when the menu is open rather than on initialization (the most performance sensitive phase). The optimization would be as follows:

```suggestion
  const handleWidthChange = () => {
    if (buttonRef.current && !isHidden) { // Oops, a double-negative - "not is hidden" means "is open", right?
      const newMenuWidth = buttonRef.current.clientWidth + 2 * buttonBorderWidth;
      setWidth(newMenuWidth);
    }
  };

  useLayoutEffect(() => {
    handleWidthChange();
  }, [buttonRef, isHidden]);
```